tools/cli/go-whisk/whisk/api.go

```diff
@@ -257,3 +338,245 @@ func (s *ApiService) Delete(api *Api, options *ApiOptions) (*http.Response, erro
+    route := "web/whisk.system/apimgmt/getApi.http"
+
+    routeUrl, err := addRouteOptions(route, apiListOptions)
+    if err != nil {
+        Debug(DbgError, "addRouteOptions(%s, %#v) error: '%s'\n", route, apiListOptions, err)
+        errMsg := wski18n.T("Unable to add route options '{{.options}}'",
+            map[string]interface{}{"options": apiListOptions})
+        whiskErr := MakeWskErrorFromWskError(errors.New(errMsg), err, EXITCODE_ERR_GENERAL, DISPLAY_MSG,
+            NO_DISPLAY_USAGE)
+        return nil, nil, whiskErr
+    }
+    Debug(DbgInfo, "Api GET/list route with api options: %s\n", routeUrl)
+
+    req, err := s.client.NewRequestUrl("GET", routeUrl, nil, DoNotIncludeNamespaceInUrl, AppendOpenWhiskPathPrefix, EncodeBodyAsJson)
+    if err != nil {
+        Debug(DbgError, "http.NewRequestUrl(GET, %s, nil, DoNotIncludeNamespaceInUrl, AppendOpenWhiskPathPrefix, EncodeBodyAsJson) error: '%s'\n", routeUrl, err)
+        errMsg := wski18n.T("Unable to create HTTP request for GET '{{.route}}': {{.err}}",
+            map[string]interface{}{"route": routeUrl, "err": err})
+        whiskErr := MakeWskErrorFromWskError(errors.New(errMsg), err, EXITCODE_ERR_NETWORK, DISPLAY_MSG,
+            NO_DISPLAY_USAGE)
+        return nil, nil, whiskErr
+    }
+
+    apiArray := new(ApiListResponseV2)
+    resp, err := s.client.Do(req, &apiArray, ExitWithErrorOnTimeout)
+    if err != nil {
+        Debug(DbgError, "s.client.Do() error - HTTP req %s; error '%s'\n", req.URL.String(), err)
+        return nil, resp, err
+    }
+
+    err = validateApiListResponse(apiArray)
+    if err != nil {
+        Debug(DbgError, "Not a valid ApiListReponse object\n")
+        return nil, resp, err
+    }
+
+    return apiArray, resp, err
+}
+
+func (s *ApiService) InsertV2(api *ApiCreateRequest, options *ApiCreateRequestOptions, overwrite bool) (*ApiCreateResponseV2, *http.Response, error) {
+    route := "web/whisk.system/apimgmt/createApi.http"
+    Debug(DbgInfo, "Api PUT route: %s\n", route)
+
+    routeUrl, err := addRouteOptions(route, options)
+    if err != nil {
+        Debug(DbgError, "addRouteOptions(%s, %#v) error: '%s'\n", route, options, err)
+        errMsg := wski18n.T("Unable to add route options '{{.options}}'",
+            map[string]interface{}{"options": options})
+        whiskErr := MakeWskErrorFromWskError(errors.New(errMsg), err, EXITCODE_ERR_GENERAL, DISPLAY_MSG,
+            NO_DISPLAY_USAGE)
+        return nil, nil, whiskErr
+    }
+    Debug(DbgError, "Api create route with options: %s\n", routeUrl)
+
+    req, err := s.client.NewRequestUrl("POST", routeUrl, api, DoNotIncludeNamespaceInUrl, AppendOpenWhiskPathPrefix, EncodeBodyAsJson)
+    if err != nil {
+        Debug(DbgError, "http.NewRequestUrl(POST, %s, nil, DoNotIncludeNamespaceInUrl, AppendOpenWhiskPathPrefix, EncodeBodyAsJson) error: '%s'\n", route, err)
```

Comment:
I think a separate issue is needed to address the consolidation of duplicate error message generation.  This PR is following the current pattern.